hack/cmd_util.sh

```diff
@@ -181,19 +198,31 @@ function os::cmd::internal::expect_exit_code_run_grep() {
 		os::text::clear_last_line
 	done
 
+	local return_code
 	if (( cmd_succeeded && test_succeeded )); then
 		os::text::print_green "SUCCESS after ${time_elapsed}s: ${name}"
+		junit_log+=( "SUCCESS after ${time_elapsed}s: ${name}" )
+
 		if [[ -n ${VERBOSE-} ]]; then
 			os::cmd::internal::print_results
 		fi
-		return 0
+		return_code=0
 	else
 		local cause=$(os::cmd::internal::assemble_causes "${cmd_succeeded}" "${test_succeeded}")
 		
 		os::text::print_red_bold "FAILURE after ${time_elapsed}s: ${name}: ${cause}"
+		junit_log+=( "FAILURE after ${time_elapsed}s: ${name}: ${cause}" )
+
 		os::text::print_red "$(os::cmd::internal::print_results)"
-		return 1
+		return_code=1
 	fi
+
+	junit_log+=( "$(os::cmd::internal::print_results)" )
+	# append inside of a subshell so that IFS doesn't get propagated out
+	( IFS=$'\n'; echo "${junit_log[*]}" >> "${JUNIT_REPORT_OUTPUT:-/dev/null}" )
```

Comment:
Once the traps PR makes it in, there will be a nice `os::util::join` function that we can use and forget about the demonic sacrifice that goes on behind the scenes to make it happen